backend/src/models/restaurant.ts

```diff
@@ -0,0 +1,22 @@
+import mongoose from 'mongoose';
+
+const menuItemsSchema = new mongoose.Schema({
+  name: { type: String, required: true },
+  price: { type: Number, required: true },
+});
+
+const restaurantSchema = new mongoose.Schema({
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/27#issuecomment-2383391092>#6fe9f8</a><span> - 09/30/2024, 02:40 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'user' field in the restaurantSchema is not marked as required, which could lead to restaurants being created without an associated user.
</div>
<div id="fix">
<b>Fix:</b> Mark the 'user' field as required in the restaurantSchema to ensure every restaurant is associated with a user.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6,7 +6,7 @@
  });

  const restaurantSchema = new mongoose.Schema({
 -  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
 +  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    restaurantName: { type: String, required: true },
    city: { type: String, required: true },
    country: { type: String, required: true },
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
good point

Reply:
updated